webhook/validation/validation.go

```diff
@@ -138,19 +146,32 @@ func hasInvalidActiveGateCapabilities(dynakube *dynatracev1beta1.DynaKube) strin
 	return ""
 }
 
-// TODO: Implement it to check other dynakubes for conflicting nodeSelectors
-//func hasConflictingNodeSelectors(dynakube dynatracev1beta1.DynaKube) bool {
-//	infraNodeSelectorMap := dynakube.Spec.InfraMonitoring.NodeSelector
-//	classicNodeSelectorMap := dynakube.Spec.ClassicFullStack.NodeSelector
-//
-//	infraNodeSelector := labels.SelectorFromSet(infraNodeSelectorMap)
-//	classicNodeSelector := labels.SelectorFromSet(classicNodeSelectorMap)
-//
-//	infraNodeSelectorLabels := labels.Set(infraNodeSelectorMap)
-//	classicNodeSelectorLabels := labels.Set(classicNodeSelectorMap)
-//
-//	return infraNodeSelector.Matches(classicNodeSelectorLabels) || classicNodeSelector.Matches(infraNodeSelectorLabels)
-//}
+func hasConflictingNodeSelector(client client.Client, dynakube *dynatracev1beta1.DynaKube) string {
+	if !dynakube.NeedsOneAgent() {
+		return ""
+	}
+	validDynakubes := &dynatracev1beta1.DynaKubeList{}
+	_ = client.List(context.TODO(), validDynakubes)
+	for i := range validDynakubes.Items {
+		nodeSelectorMap := dynakube.NodeSelector()
+		validNodeSelectorMap := validDynakubes.Items[i].NodeSelector()
+		if hasConflictingMatchLabels(nodeSelectorMap, validNodeSelectorMap) {
+			return fmt.Sprintf(errorNodeSelectorConflict, validDynakubes.Items[i].Name)
+		}
+	}
+	return ""
+}
+
+func hasConflictingMatchLabels(labelMap1, labelMap2 map[string]string) bool {
```

Comment:
I would prefer names like `labelMap` and `otherLabelMap`, since the first map is referring to the current Dynakube